hack/dev/test-appset.yml

```diff
@@ -0,0 +1,35 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: appset-goinfra
+  namespace: argocd
+spec:
+  generators:
+  - list:
+      elements:
+        - cluster: control-plane
+          url: https://kubernetes.default.svc
+        - cluster: kind-prc-cluster-1
+          url: https://prc-cluster-1-control-plane:6443
+        - cluster: kind-prc-cluster-2
+          url: https://prc-cluster-2-control-plane:6443
+  template:
+    metadata:
+      name: 'goinfra-{{cluster}}'
+    spec:
+      project: default
+      source:
+        repoURL: https://github.com/maruina/appset-poc.git
```

Comment:
Should we point this to this repository instead of relying on a third party one?

Reply:
Can this be left for latter iterations? This PR is already getting a bit large, and I wouldn't want to add additional files to it at the moment.